runtime/src/system_instruction_processor.rs

```diff
@@ -1,13 +1,45 @@
 use log::*;
-use solana_sdk::account::KeyedAccount;
-use solana_sdk::instruction::InstructionError;
-use solana_sdk::instruction_processor_utils::{limited_deserialize, next_keyed_account};
-use solana_sdk::pubkey::Pubkey;
-use solana_sdk::system_instruction::{SystemError, SystemInstruction};
-use solana_sdk::system_program;
-use solana_sdk::sysvar;
-
-fn create_system_account(
+use solana_sdk::{
+    account::KeyedAccount,
+    instruction::InstructionError,
+    instruction_processor_utils::{limited_deserialize, next_keyed_account},
+    pubkey::Pubkey,
+    system_instruction::{create_address_with_seed, SystemError, SystemInstruction},
+    system_program, sysvar,
+};
+
+fn create_account_with_seed(
+    from: &mut KeyedAccount,
+    to: &mut KeyedAccount,
+    seed: &str,
```

Comment:
Any reason for the restriction to 32B ASCII string? `[u8; 32]` could support some internationalization and hash digests

Reply:
yeah good point, we don't care that the seed is valid utf-8

Reply:
I wanted to keep the space smaller than 256 bytes of seed to deter grinding attacks.  32 ascii chars takes away 32 bits.

Reply:
@rob-solana grinding on what though?

Reply:
trying to DoS a guy you think wants pubkey+"my secret stake \#23"

Reply:
if we think 256 bits of seed is ok, I'll just accept another Hash instead of an ascii string.

Reply:
@rob-solana you can always grind with pubkey generation.  I think we should either have a Hash/[u8;32], or u64.

Reply:
Requiring `from`'s signature effectively makes griefing the address space a second-preimage attack on SHA256.  We should probably be OK with that.  Though fewer bits _do_ minimize opportunities for shenanigans.  Would it make sense to restrict the seed to something as small as a `u16`?

Reply:
were it as small as a u16, mnemonics get harder.   ascii is nice for mnemonics (in English).  opening to utf-8 would make it look less amateur-hour.

The grinding attack is picking another source `from` (obs having the keypair), and griefing some other dude's `from`.  The goal is to make that griefing look hard or futile.  

I think we can do this with:
1. u64, maybe low 8 bytes of hash(mnemonic string)?
2. hashing a limited-length string as proposed.  Upgrading to utf-8 hurts the deterrent a little as you can cover 256 bits of space with only 13-ish utf-8 chars (at 20bits/char).

We can also say "who cares?  grind away!"

Reply:
I'm planning to remove the ascii stuff, appreciate input here @t-nelson @aeyakovenko @garious

Reply:
> were it as small as a u16, mnemonics get harder. ascii is nice for mnemonics (in English). opening to utf-8 would make it look less amateur-hour.

Right, the `u16` suggestion would discard the mnemonic for an index (`from` gets `I` accounts under `program_id`). Which gives cheap introspection, but no pseudonymity.

> The grinding attack is picking another source `from` (obs having the keypair), and griefing some other dude's `from`. The goal is to make that griefing look hard or futile.

Yep. A SHA256 second preimage search is pretty futile IMO.  Restricting free bits per `from` is about the most effective deterrent we can enact.  Grinding `from`s is always possible and slower than the hashing step.  You've already covered the other speedup.  That is, requiring fixed bits at the end of the input data, preventing working from a midstate.
  
> I think we can do this with:
> 
>     1. u64, maybe low 8 bytes of hash(mnemonic string)?
> 
>     2. hashing a limited-length string as proposed.  Upgrading to utf-8 hurts the deterrent a little as you can cover 256 bits of space with only 13-ish utf-8 chars (at 20bits/char).

I think (valid) utf-8 would actually be _more_ restricted.  For non-ASCII code points, the top *two* bits of every byte are fixed and the first byte also consumes up to three more bits to encode the length.  Not all code pages are in use or full either, further restricting valid bitstrings

> We can also say "who cares? grind away!"

Probably fine. See, https://lbc.cryptoguru.org/about